zebra-consensus/src/block/check.rs

```diff
@@ -144,22 +144,23 @@ pub fn equihash_solution_is_valid(header: &Header) -> Result<(), equihash::Error
     header.solution.check(header)
 }
 
-/// Returns `Ok(())` if the block subsidy in `block` is valid for `network`
+/// Returns `Ok()` with the deferred pool balance change of the coinbase transaction if
+/// the block subsidy in `block` is valid for `network`
 ///
 /// [3.9]: https://zips.z.cash/protocol/protocol.pdf#subsidyconcepts
 pub fn subsidy_is_valid(
     block: &Block,
     network: &Network,
     expected_block_subsidy: Amount<NonNegative>,
-) -> Result<(), BlockError> {
+) -> Result<Amount, BlockError> {
```

Comment:
nit: Since this is a value with specific protocol semantics, I might prefer a newtype to a bare `Amount` to make the returned value harder to misuse.

Reply:
Applied in https://github.com/ZcashFoundation/zebra/pull/9603/commits/aafe3e568d89f8940d42ed0ee294fcfcc45e4d2b.